llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -3424,7 +3424,15 @@ InstructionCost AArch64TTIImpl::getGatherScatterOpCost(
   // Add on an overhead cost for using gathers/scatters.
   // TODO: At the moment this is applied unilaterally for all CPUs, but at some
   // point we may want a per-CPU overhead.
-  MemOpCost *= getSVEGatherScatterOverhead(Opcode);
+  unsigned OpCost = 1;
+  if (ST->getProcFamily() == AArch64Subtarget::NeoverseV2) {
+     // Specialize overhead of scatter instructions on Neoverse-V2
+    if (Opcode == Instruction::Store)
+      OpCost = 13;
```

Comment:
What happens if `Opcode != Instruction::Store`?

Reply:
I restructed the code a bit.